src/storage/mapping/FixedContentTypeMapper.ts

```diff
@@ -21,4 +45,35 @@ export class FixedContentTypeMapper extends BaseFileIdentifierMapper {
   protected async getContentTypeFromPath(): Promise<string> {
     return this.contentType;
   }
+
+  public async mapUrlToDocumentPath(identifier: ResourceIdentifier, filePath: string, contentType?: string):
+  Promise<ResourceLink> {
+    // Handle URL suffix
+    if (this.urlSuffix) {
+      if (filePath.endsWith(this.urlSuffix)) {
+        filePath = filePath.slice(0, filePath.length - this.urlSuffix.length);
+      } else {
+        this.logger.error(`Trying to access URL ${filePath} outside without required suffix ${this.urlSuffix}`);
+        throw new NotFoundHttpError(
+          `Trying to access URL ${filePath} outside without required suffix ${this.urlSuffix}`,
+        );
+      }
+    }
+
+    return super.mapUrlToDocumentPath(identifier, filePath + this.pathSuffix, contentType);
+  }
+
+  protected async getDocumentUrl(relative: string): Promise<string> {
+    // Handle path suffix
+    if (this.pathSuffix) {
+      if (relative.endsWith(this.pathSuffix)) {
+        relative = relative.slice(0, relative.length - this.pathSuffix.length);
```

Comment:
Same here. Wondering if these can be combined into a same function but perhaps might make the logger/error text too generic.

Reply:
Feels a bit overkill to abstract this, given that the warning/error messages take up most of the characters in this block.